crates/metadata/src/specs.rs

```diff
@@ -366,21 +393,25 @@ impl<S, P> ConstructorSpecBuilder<S, P> {
     /// Sets the documentation of the message specification.
     pub fn docs<D>(self, docs: D) -> Self
     where
-        D: IntoIterator<Item = &'static str>,
+        D: IntoIterator<Item = <F as Form>::String>,
+        F::String: DocString,
     {
         let mut this = self;
         debug_assert!(this.spec.docs.is_empty());
         this.spec.docs = docs
             .into_iter()
-            .map(trim_extra_whitespace)
+            .map(DocString::trim_extra_whitespace)
```

Comment:
```suggestion
            .map(|s| trim_extra_whitespace(s.as_ref()))
```
Then you *should* be able to do something like this.